test/test_simulation.c

```diff
@@ -0,0 +1,90 @@
+}
+
+void test_null_cjson(void) {
+    int res = build_simulation_scenario(NULL, NULL, NULL);
+    TEST_ASSERT_EQUAL(SIM_CONFIG_EMPTY, res);
+}
+
+void test_invalid_model_selection(void) {
+    cJSON *json = NULL;
+
+    read_json_file("./data/simulation/2_1_empty_config.json", &json);
+    TEST_ASSERT_EQUAL(SIM_MODEL_SELECTION_INVALID, build_simulation_scenario(json, NULL, NULL));
+
+    read_json_file("./data/simulation/2_2_nonstring_model.json", &json);
+    TEST_ASSERT_EQUAL(SIM_MODEL_SELECTION_INVALID, build_simulation_scenario(json, NULL, NULL));
+
+    read_json_file("./data/simulation/2_3_nonexisting_model.json", &json);
+    TEST_ASSERT_EQUAL(SIM_MODEL_SELECTION_INVALID, build_simulation_scenario(json, NULL, NULL));
+}
+
+void test_invalid_default_model_common_config(void) {
+    cJSON *json = NULL;
+
+    read_json_file("./data/simulation/3_1_no_default_config_model.json", &json);
+    TEST_ASSERT_EQUAL(SIM_MODEL_COMMON_CONFIG_INVALID, build_simulation_scenario(json, NULL, NULL));
+
+    read_json_file("./data/simulation/3_2_nonobject_default_config_model.json", &json);
+    TEST_ASSERT_EQUAL(SIM_MODEL_COMMON_CONFIG_INVALID, build_simulation_scenario(json, NULL, NULL));
+
+    read_json_file("./data/simulation/3_3_empty_default_config_model.json", &json);
+    TEST_ASSERT_EQUAL(SIM_MODEL_COMMON_CONFIG_INVALID, build_simulation_scenario(json, NULL, NULL));
+
+    read_json_file("./data/simulation/3_4_incomplete_default_config_model.json", &json);
+    TEST_ASSERT_EQUAL(SIM_MODEL_COMMON_CONFIG_INVALID, build_simulation_scenario(json, NULL, NULL));
+
+    read_json_file("./data/simulation/3_5_invalid_delay_default_config_model.json", &json);
+    TEST_ASSERT_EQUAL(SIM_MODEL_COMMON_CONFIG_INVALID, build_simulation_scenario(json, NULL, NULL));
+}
+
+void test_invalid_default_model_specific_config(void) {
+    cJSON *json = NULL;
+
+    read_json_file("./data/simulation/4_1_empty_specific_config_model.json", &json);
+    TEST_ASSERT_EQUAL(SIM_MODEL_SPECIFIC_CONFIG_INVALID, build_simulation_scenario(json, NULL, NULL));
+
+    read_json_file("./data/simulation/4_2_invalid_specific_config_params_model.json", &json);
+    TEST_ASSERT_EQUAL(SIM_MODEL_SPECIFIC_CONFIG_INVALID, build_simulation_scenario(json, NULL, NULL));
+
+    read_json_file("./data/simulation/4_3_missing_specific_config_params_model.json", &json);
+    TEST_ASSERT_EQUAL(SIM_MODEL_SPECIFIC_CONFIG_INVALID, build_simulation_scenario(json, NULL, NULL));
+}
+
+void test_invalid_output_config_path(void) {
+    cJSON *json = NULL;
+
+    read_json_file("./data/simulation/5_1_missing_output_config_path.json", &json);
+    TEST_ASSERT_EQUAL(SIM_CONFIG_OUTPUT_PATH_INVALID, build_simulation_scenario(json, NULL, NULL));
+
+    read_json_file("./data/simulation/5_2_invalid_output_config_path.json", &json);
+    TEST_ASSERT_EQUAL(SIM_CONFIG_OUTPUT_PATH_INVALID, build_simulation_scenario(json, NULL, NULL));
+
+    read_json_file("./data/simulation/5_3_existing_output_config_path.json", &json);
```

Comment:
I think the path in this .json file must also be changed. Or the error code must be changed.